data/code/lib.c

```diff
@@ -323,3 +324,170 @@ int itemshufflemgr_calcslot_ex(const unsigned int* const itemshufflemgr,
 }
 
 #endif
+
+#if SECTIONED_COURSES
+
+static unsigned short section_count = 0;
+
+// Due to the nature of the compiler, a lot of the code had to be rewritten in ASM
+// so that the compiler would not ignore it, and thus break this code patch.
+// To compensate, nearly every set of ASM instructions has a description of what it's doing.
+
+// Override the lap count in a section course to be the number of section points.
+void section_lap_override()
+{
+    register unsigned short reg9 asm("r9");
+    asm("or %r0,%r3,%r3 \t\n");
```

Comment:
Nit:

```diff
- asm("or %r0,%r3,%r3 \t\n");
+ asm("or %r0, %r3, %r3");
```

Generally, I think the assembly code in this PR could do with some consistency. At the moment, whitespace characters are all over the place. Sometimes there are multiple spaces after the instruction name, sometimes there are no spaces between operands, sometimes there are a bunch of trailing spaces, as well as that tab (`\t`) character.

My preference:
- A single space between instruction name and first operand (unless we want to align instructions and operands within the same `asm()` block, which I'd be okay with).
- A single space between operands.
- Where applicable, no space between offset and GPR.
- No trailing whitespace character, except for `\n`, when more than one line is present in the assembly block. We could consider `;` instead of `\n`, actually: it'd probably be more visually appealing overall.